src/Symfony/Component/DependencyInjection/Compiler/AutowirePass.php

```diff
@@ -525,7 +525,16 @@ private static function getResourceMetadataForMethod(\ReflectionMethod $method)
         $methodArgumentsMetadata = [];
         foreach ($method->getParameters() as $parameter) {
             try {
-                $class = $parameter->getClass();
+                if (method_exists($parameter, 'getType')) {
+                    $type = $parameter->getType();
```

Comment:
do we need to call hasType before? what value should $class have when a parameter has no type? false or null?

Reply:
> do we need to call hasType before?

Not necessarily. `getType()` will return null if there is no type declaration.

> what value should $class have when a parameter has no type? false or null?

The previous logic was:
* `null`: No parameter type or type is not a class.
* `false`: Type is a class, but we were unable to reflect on it.